controller.js

```diff
@@ -45,3 +46,18 @@ exports.getArticles = (req, res, next) => {
     });
   });
 };
+
+exports.getComments = (req, res, next) => {
+  const articleID = req.params.article_id;
+  if (!isNaN(Number(articleID))) {
+    selectComments(articleID)
+      .then((selectedComments) => {
+        res.status(200).send(selectedComments);
+      })
+      .catch((err) => {
+        next();
+      });
+  } else {
+    next("invalid id");
+  }
```

Comment:
We should not be trying to handle the error in the controller. This should ideally be done in the model file. The reason for this is so we have one .catch which deals with all errors. Currently you are invoking next in two different areas which isnt ideal.

This type of error can also be left for the query itself to deal with as the sql query itself will fail and throw an error itself that you can try and deal with in your middleware error handlers.